insights/parsers/ntp_sources.py

```diff
@@ -84,26 +101,43 @@ class NtpqLeap(CommandParser):
 
     Examples:
 
-        >>> print shared[NtpqLeap].leap  # same data
+        >>> type(ntpq)
+        <class 'insights.parsers.ntp_sources.NtpqLeap'>
+        >>> ntpq.leap
         '00'
     """
 
     def parse_content(self, content):
-        if "Connection refused" in content[0]:
+        if content and "Connection refused" in content[0]:
             raise SkipComponent("NTP service is down and connection refused")
-        self.data = {}
+
+        leap = None
         for line in content:
-            m = re.search(r'leap=(\d*)', line)
-            if m:
-                self.data["leap"] = m.group(1)
+            if 'leap=' in line:
+                leap = line.split('leap=')[1].split()[0]
```

Comment:
@xiangce it might be clearer that you are stripping the line ending here by using `rstrip` here instead.